r/R/arrow-package.R

```diff
@@ -196,6 +196,20 @@ configure_tzdb <- function() {
         )
       )
     }
+
+    # On MacOS only, Check if we are running in under emulation, and warn this will not work
+    rosetta <- identical(sysname, "darwin") && identical(system("sysctl -n sysctl.proc_translated", intern = TRUE), "1")
+    if (rosetta) {
+      packageStartupMessage(
```

Comment:
Should this actually be a `warning()`? Or even a `stop()`?

Reply:
I guess it depends on the extent of the problem; does it always segfault, or just when we use some of the functionality?

Reply:
There might be some functionality that works, but everything I've seen people try (even `open_dataset() |> nrow()` segfaults. Given opcodes + architectures + emulation, I would be surprised if much of anything of use in arrow will work.

Reply:
I think `packageStartupMessage` is appropriate, I wouldn't raise other signals in onLoad